src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs

```diff
@@ -19,6 +19,7 @@ public static partial class JsonSerializer
             ref WriteStack state)
         {
             bool finishedSerializing;
+            int currentDepth = writer.CurrentDepth;
```

Comment:
nit: I would rename this to be `originalDepth`
```suggestion
            int originalDepth = writer.CurrentDepth;
```

Reply:
Haha, I meant to do that but was too excited to make the PR